AK/MemoryStream.cpp

```diff
@@ -270,13 +270,19 @@ ErrorOr<Bytes> AllocatingMemoryStream::next_write_range()
 
 void AllocatingMemoryStream::cleanup_unused_chunks()
 {
-    // FIXME: Move these all at once.
-    while (m_read_offset >= CHUNK_SIZE) {
+    auto const chunks_to_remove = m_read_offset / CHUNK_SIZE;
+    auto const chunks_to_move = m_chunks.size() - chunks_to_remove;
+
+    if (chunks_to_remove > 0) {
         VERIFY(m_write_offset >= m_read_offset);
 
-        auto buffer = m_chunks.take_first();
-        m_read_offset -= CHUNK_SIZE;
-        m_write_offset -= CHUNK_SIZE;
+        for (u64 i = 0; i < chunks_to_move; ++i)
+            m_chunks[i] = move(m_chunks[i + chunks_to_remove]);
+
+        m_chunks.resize(chunks_to_move);
```

Comment:
On second look, is this whole function just manually doing `Vector::remove(index, size)`? i.e. is this (or some adjustment to this) equivalent:
```c++
void AllocatingMemoryStream::cleanup_unused_chunks()
{
    if (auto chunks_to_remove = m_read_offset / CHUNK_SIZE; chunks_to_remove > 0) {
        m_chunks.remove(0, chunks_to_remove);
        m_read_offset -= CHUNK_SIZE * chunks_to_remove;
        m_write_offset -= CHUNK_SIZE * chunks_to_remove;
    }
}
```